quantfinlib/train_test_split/walk_forward.py

```diff
@@ -0,0 +1,114 @@
+"""Implementation of Walk forward."""
+
+from numbers import Integral
+from typing import Generator, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+
+from quantfinlib.train_test_split._base import BaseCV, _embargo
+
+
+class WalkForward(BaseCV):
+    """Walk forward train-test split."""
+
+    def __init__(
+        self,
+        n_embargo: Union[Integral, pd.Timedelta],
+        n_purge: Union[Integral, pd.Timedelta] = 0,
+        train_window: Union[Integral, pd.Timedelta] = 0,
+        test_window: Union[Integral, pd.Timedelta] = 0,
+        wf_split_type: str = "expanding",
+        freq: Optional[str] = None,
+    ) -> None:
+        super().__init__(n_embargo=n_embargo, n_purge=n_purge)
+        self.train_window = train_window
+        self.test_window = test_window
+        self.wf_split_type = wf_split_type
+        if wf_split_type not in ["expanding", "rolling"]:
+            raise ValueError("Invalid value for 'wf_split_type'. Must be either 'expanding' or 'rolling'.")
+        self.freq = freq
+        self.n_splits = 0
+        self._finalized_splits = False
+
+    def __repr__(self):

```

Comment:
I remember that __repr__ is for developers, __str__ is for users, but in case there is no __str__ any call of str(obj) will fall back to __repr__, so it is fine.